src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -356,13 +360,10 @@ public void setChassisSpeeds(ChassisSpeeds chassisSpeeds, boolean isOpenLoop) {
    * 
    * Resets the pigeon IMU's yaw.
    * 
-   * @param degrees the new yaw angle, in degrees.
+   * @param rotation the new yaw angle as Rotation2d
    */
-  public void setYaw(double degrees) {
-    // the odometry stores an offset from the current pigeon angle
-    // changing the angle makes that offset inaccurate, so must reset the pose as well.
-    // keep the same translation, but set the odometry angle to what we want the angle to be.
-    resetOdometry(new Pose2d(getPose().getTranslation(), Rotation2d.fromDegrees(degrees)));
+  public void setYaw(Rotation2d rotation) {
+    m_pigeon.setYaw(rotation.getDegrees());
```

Comment:
It is very confusing to have public methods setYaw(), setPigeonYaw(), getYaw() and getRawPigeonYaw() methods with slightly different semantics and no clear explanation of what they do. Can we reduce it to just a pair of public setXX/getXX? E.g., the setPigeonYaw could be called setYawForAlllance and moved to the caller or utility class.